File: SingleAPIExample/python/Refund.py
import splitit

username, password, api_key = (
    "username",
    "password",
    "api_key",
)

api_client = splitit.ApiClient(splitit.Configuration(username, password, api_key, sandbox=True))


def refund_installment_plan(ipn):
    resp = api_client.InstallmentPlanApi.installment_plan_refund(
        splitit.RefundPlanRequest(
            installment_plan_number=ipn,
            amount=splitit.MoneyWithCurrencyCode(200, "USD"),
            refund_strategy=splitit.RefundStrategy.FUTUREINSTALLMENTSFIRST,
        )
    )
    return resp

File: SingleAPIExample/java/Refund.java
import com.splitit.*;
import com.splitit.sdk.model.*;
import java.math.BigDecimal;


public class Refund {
    private String apiKey, username, password;
    public Refund(String username, String password, String apiKey){
        this.username = username;
        this.password = password;
        this.apiKey = apiKey;
    }

    public void example(String ipn) throws ApiException{
        ApiClient apiClient = new ApiClient()
            .sandbox(true)
            .username(username)
            .password(password)
            .apiKey(apiKey);


        var result = apiClient.getInstallmentPlanApi().installmentPlanRefund(
            new RefundPlanRequest()
            .installmentPlanNumber(ipn)
            .amount(
                new MoneyWithCurrencyCode()
                .value(new BigDecimal(200))
                .currencyCode("USD")
            )
            .refundStrategy(RefundStrategy.FUTUREINSTALLMENTSFIRST)

        );
        System.out.println(result);
    }
}
